lib/_stream_writable.js

```diff
@@ -248,10 +248,18 @@ Writable.prototype.pipe = function() {
 
 
 function writeAfterEnd(stream, cb) {
-  const er = new ERR_STREAM_WRITE_AFTER_END();
-  // TODO: defer error events consistently everywhere, not just the cb
-  errorOrDestroy(stream, er);
-  process.nextTick(cb, er);
+  // TODO: defer error events consistently everywhere, not just the cb.
+  if (stream._writeAfterEnd) {
+    stream._writeAfterEnd(err => {
+      const er = err || new ERR_STREAM_WRITE_AFTER_END();
+      errorOrDestroy(stream, er);
+      process.nextTick(cb, er);
+    })
+  } else {
+    const er = new ERR_STREAM_WRITE_AFTER_END();
+    errorOrDestroy(stream, er);
+    process.nextTick(cb, er);
+  }
```

Comment:
I think we should document & test this method, as it can be overridden by implementors.

Reply:
added test and docs